### PR TITLE
MODORDSTOR-80 basic CRUD for relationship between order and invoice 

### PR DESCRIPTION
**Pull request by RamanAsadchy**
## Purpose
Implement CRUD operations for relationship between order and invoice

## Learning

@folio-org/acquisitions

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: ramls/order_invoice_relationship.raml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  type:
+    collection:
+      exampleCollection: !include acq-models/mod-orders-storage/examples/order_invoice_relationship_collection.sample
+      exampleItem: !include acq-models/mod-orders-storage/examples/order_invoice_relationship_post.sample
+      schemaCollection: order-invoice-relationship-collection
+      schemaItem: order-invoice-relationship
+  get:
+    description: Get list of relationships
+    is: [
+      searchable: {description: "with valid searchable fields: for example code", example: "[\"code\", \"MEDGRANT\", \"=\"]"},
```
Comment (grabsefx) [inline ramls/order_invoice_relationship.raml]: Please, use valid example and description of searchable trait.  `order_invoice_relationship` contains only two fields: `invoiceId` and `purchaseOrderId`, there is no `code` field.

File: descriptors/ModuleDescriptor-template.json [hunk 2]
```diff
@@ -200,6 +200,37 @@
         }
       ]
     },
+    {
+      "id": "orders-storage.order-invoice-relationship",
```
Comment (grabsefx) [inline descriptors/ModuleDescriptor-template.json]: I think, interface naming "orders-storage.order-invoice-relationship**s**" should be used. Also permissions naming should be changed to use "order-invoice-relationship**s**"

Comment (RamanAsadchy) [reply descriptors/ModuleDescriptor-template.json]: Done. Check again please.

File: ramls/order_invoice_relationship.raml (hunk 1 above)
Comment (RamanAsadchy) [reply ramls/order_invoice_relationship.raml]: Done. Check again please.

File: src/main/resources/templates/db_scripts/schema.json [hunk 3]
```diff
@@ -64,6 +64,19 @@
           "targetTable": "po_line"
         }
       ]
+    },
+    {
+      "tableName": "order_invoice_relationship",
+      "generateId": true,
+      "fromModuleVersion": 6.0,
+      "withMetadata": true,
```
Comment (piotr-kalashuk) [inline src/main/resources/templates/db_scripts/schema.json]: The [order_invoice_relationship.json](https://github.com/folio-org/acq-models/blob/master/mod-orders-storage/schemas/order_invoice_relationship.json) schema does not include metadata so `withMetadata` should be `false`

**Pull request merged**